### PR TITLE
memkind: Conflicts with jemalloc

### PR DESCRIPTION
**Pull request by eschnett**

### CODE REVIEW

**Review by alalazo — CHANGES_REQUESTED** [review]
Is there any way to use an external `jemalloc`?

**Comment by eschnett** [comment]
I don't know, I am not an expert in this package. I do not see such an option in the `configure` file.

`memkind` renames the public jemalloc symbols, but some private symbols (e.g. `je_malloc_default`) are not renamed, and are still exported by the static memkind library.

File: var/spack/repos/builtin/packages/memkind/package.py [hunk 1]
```diff
@@ -38,6 +38,10 @@ class Memkind(AutotoolsPackage):
     depends_on('m4',       type='build')
     depends_on('numactl')
 
+    # memkind includes a copy of jemalloc.
+    # (It really shouldn't; with Spack, it should use an external jemalloc.)
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/memkind/package.py]: Can you rework this comment to point to this link: https://github.com/memkind/memkind#jemalloc ? Ideally I think we should patch `jemalloc` when building `memkind` but I have no idea of the amount of work that would require (so I'm fine merging this once we have the link in the comment)

**Review by alalazo — CHANGES_REQUESTED** [review]

File: var/spack/repos/builtin/packages/memkind/package.py (hunk 1 above)
Comment (eschnett) [reply var/spack/repos/builtin/packages/memkind/package.py]: Done.

**Review by alalazo — APPROVED** [review]

**Pull request merged**